src/brod.erl

```diff
@@ -891,20 +1153,71 @@ fetch(ConnOrBootstrap, Topic, Partition, Offset) ->
   fetch(ConnOrBootstrap, Topic, Partition, Offset, Opts).
 
 %% @doc Fetch a single message set from the given topic-partition.
+%%
 %% The first arg can either be an already established connection to leader,
-%% or `{Endpoints, ConnConfig}' so to establish a new connection before fetch.
+%% or `{Endpoints, ConnConfig}' (or just `Endpoints') so to establish a new
+%% connection before fetch.
+%%
+%% The fourth argument is the start offset of the query. Messages with offset
+%% greater or equal will be fetched.
+%%
+%% You can also pass options for the fetch query.
+%% See the {@link kpro_req_lib:fetch_opts()} type for the list of them.
+%% Their meaning is documented
+%% <a href="https://tinyurl.com/kafka-fetch-api-doc">here</a>.
+%% Note that `max_bytes' will be rounded up so that full messages are
+%% retrieved. For example, if you specify `max_bytes = 42' and there
+%% are three messages of size 40 bytes, two of them will be fetched.
+%%
+%% On success, the function returns the messages along with the <i>last stable
+%% offset</i> (when using `read_committed' mode, the last committed offset) or the
+%% <i>high watermark offset</i> (offset of the last message that was successfully
+%% copied to all replicas, incremented by 1), whichever is lower. In essence, this
+%% is the offset up to which it was possible to read the messages at the time of
+%% fetching. This is similar to what {@link resolve_offset/6} with `latest'
+%% returns. You can use this information to determine how far from the end of the
+%% topic you currently are. Note that when you use this offset as the start offset
+%% for a subseuqent call, an empty list of messages will be returned (assuming the
+%% topic hasn't changed, e.g. no new message arrived). Only when you use an offset
+%% greater than this one, `{error, offset_out_of_range}' will be returned.
+%%
+%% Note also that the messages are fetched only up to the end of a topic segment
+%% in which the first retrieved message is, so there may actually be more
+%% messages in the topic than were retrieved (even in a static topic).
```

Comment:
not sure what a 'static topic' is, no producers appending messages while fetching ?
```suggestion
%% messages behind the last fetch offset even if the fetch size is significantly less than `max_bytes` provided in `fetch_opts()`.
```

Reply:
> not sure what a 'static topic' is, no producers appending messages while fetching ?

Yes, this was what I meant by *static topic*.

Thank you for making it more clear. I am not sure what you mean by *fetch size* though.

Reply:
Aha, it makes sense now. I'll use your suggestion and commit it locally.